src/widgets/medications/medications-overview.component.tsx

```diff
@@ -202,6 +173,43 @@ export default function MedicationsOverview(props: MedicationsOverviewProps) {
       );
     });
   }
+
+  function displayMedications() {
```

Comment:
**For all places where you are using this pattern**
1. Main component vs supporting component: It seems to be the norm in react that the JSX bit of the component goes at the end of the function component as a return statement. The supporting components(which is consumed by the main component) can be declare inline, or in separate files. The component in question here is the 'no content' component which appears to be the main component of the function. 
2. Use of lambda expressions to avoid running into bugs: you can avoid the risk of bugs due to 'this' reference by using lambda expressions as:
`const noMedications = () => (<><>)`
3 You are repeating yourself a lot. Maybe that indicates that you need to refactor the 'no content' component into a hook/function component?